File: scratch/remove_bg.js
const { Jimp } = require('jimp');

async function removeBackground() {
  try {
    const img = await Jimp.read('public/brand/floating-icons.png');
    
    // Iterate over all pixels
    img.scan(0, 0, img.bitmap.width, img.bitmap.height, function (x, y, idx) {
      const r = this.bitmap.data[idx + 0];
      const g = this.bitmap.data[idx + 1];
      const b = this.bitmap.data[idx + 2];
      
      // If the pixel is close to white, make it transparent
      if (r > 240 && g > 240 && b > 240) {
        this.bitmap.data[idx + 3] = 0; // Set alpha to 0
      }
    });
    
    await img.write('public/brand/floating-icons-transparent.png');
    console.log("Background removed successfully!");
  } catch (err) {
    console.error("Error:", err);
  }
}

removeBackground();

File: scratch/remove_bg.py
from PIL import Image

def remove_background(input_path, output_path):
    img = Image.open(input_path)
    img = img.convert("RGBA")
    
    datas = img.getdata()
    
    new_data = []
    for item in datas:
        # If the pixel is close to white, make it transparent
        # (255, 255, 255) is pure white
        if item[0] > 240 and item[1] > 240 and item[2] > 240:
            new_data.append((255, 255, 255, 0))
        else:
            new_data.append(item)
            
    img.putdata(new_data)
    img.save(output_path, "PNG")

if __name__ == "__main__":
    remove_background("c:/website2/nckh2-clone/public/brand/floating-icons.png", "c:/website2/nckh2-clone/public/brand/floating-icons.png")
